chrome/background.js

```diff
@@ -157,8 +151,177 @@ getStorageValue('extension_running').then(extensionRunning => {
 });
 
 // User switches between tabs
-chrome.tabs.onActivated.addListener(function(activeInfo) {
+chrome.tabs.onActivated.addListener(function (activeInfo) {
   chrome.tabs.get(activeInfo.tabId, (tab) => {
     handleTabChange(tab);
   });
-});
\ No newline at end of file
+});
+
+
+/* Request intercepting */
+chrome.webRequest.onBeforeRequest.addListener(
+  onBeforeRequest, { urls: ["<all_urls>"] }, ['blocking']);
+
+chrome.webRequest.onBeforeRedirect.addListener(
+  onBeforeRedirect, { urls: ["<all_urls>"] });
+
+chrome.webRequest.onErrorOccurred.addListener(
+  onErrorOccurred, { urls: ["<all_urls>"] });
+
+function onBeforeRequest(requestInfo) {
+  console.log("request" + JSON.stringify(requestInfo));
+
+  const url = new URL(requestInfo.url);
+
+  // Skip all weird requests...
+  if (url.hostname == "localhost") {
+    return {}
+  }
+
+  if (requestInfo.url.startsWith("chrome-extension")) {
+    return {};
+  }
+
+  if (requestInfo.initiator && requestInfo.initiator.startsWith("chrome-extension")) {
+    return {};
+  }
+
+  const requestDBEntry = {
+    requestId: requestInfo.requestId,
+    tabId: requestInfo.tabId,
+    domain: url.hostname,
+    mainDomain: requestInfo.initiator ? new URL(requestInfo.initiator).hostname : '',
+  };
+
+
+  // If we don't have any information about scion-enabled or not
+  if (!knownNonSCION[url.hostname] && !knownSCION[url.hostname]) {
+    // TODO: We could do this also in the onErrorOccured/onBeforeRedirect...
```

Comment:
Do we need this if we already have the onErrorOccured/onBeforeRedirect hack?

Reply:
This was quite interesting. I tried to do it in this hack but it didn't work. I wrote a comment why.